src/Compilers/CSharp/Test/Symbol/Compilation/CompilationAPITests.cs

```diff
@@ -2053,6 +2053,30 @@ public void PreviousSubmissionWithError()
             Assert.Throws<InvalidOperationException>(() => CreateSubmission("a + 1", previous: s0));
         }
 
+        [Fact]
+        public void CreateTupleTypeSymbol()
+        {
+            var comp = CSharpCompilation.Create("test", references: new[] { MscorlibRef } );
+            ITypeSymbol intType = comp.GetSpecialType(SpecialType.System_Int32);
+            ITypeSymbol stringType = comp.GetSpecialType(SpecialType.System_String);
+
+            Assert.Throws<ArgumentNullException>(() => comp.CreateTupleTypeSymbol(default(ImmutableArray<ITypeSymbol>), default(ImmutableArray<string>)));
+
+            Assert.Throws<ArgumentException>(() => comp.CreateTupleTypeSymbol(ImmutableArray<ITypeSymbol>.Empty, default(ImmutableArray<string>)));
+            Assert.Throws<ArgumentException>(() => comp.CreateTupleTypeSymbol(new[] { intType }.AsImmutable(), default(ImmutableArray<string>)));
+
+            Assert.Throws<ArgumentException>(() => comp.CreateTupleTypeSymbol(new[] { intType, intType }.AsImmutable(), new[] { "Item1" }.AsImmutable()));
```

Comment:
What about the compiler scenario where we use reserved names like Item1?  This test doesn't seem to cover that, or at least it's misleading, because it's hitting a different case: tuple type count and element count differ.